### ISSUE TITLE
Eprop - Non-finite detected in hidden parameters gradients

### ISSUE THREAD

**Issue by AnthoDrouin** [NeuroTorch/NeuroTorch#127]
When running Eprop, one might obtain the following error message from NeuroTorch :

`Non-finite detected in hidden parameters gradients`

This error is used to signal that the parameters contained non finite elements such as inf or NaN. This seems to only be problematic when the learning rate and/or the weight decay is increased. This is true even when there is clipping. Hence it seems that clipping does not prevent totally the parameters from diverging.

When using the debugger, we can see that the parameters might become NaN :
<img width="706" alt="image" src="https://user-images.githubusercontent.com/93488840/220669104-f7e41b76-f69a-421f-8b4b-79ced5972a76.png">

The error is detected in Eprop :

<img width="766" alt="image" src="https://user-images.githubusercontent.com/93488840/220669475-c5d5bfbb-122a-4217-9461-41fb2b423452.png">
